__tests__/nc_news_test.test.js

```diff
@@ -181,31 +182,63 @@ describe('GET /api/topics', () => {
-            return request(app)
-            .get("/api/articles/7/comments")
-            .expect(404)
-            .then((response)=> {
-                expect(response.body.msg).toBe("Not Found")
-            })
         });
         test('GET /api/articles/:article_id/comments responds with \'Invalid Input\' message and 400 code, when requested with the wrong input type for ID', () => {
             return request(app)
             .get("/api/articles/Seven/comments")
             .expect(400)
             .then((response)=> {
-                expect(response.body.msg).toBe("Invalid Input")
+                expect(response.body.msg).toBe("Bad Request")
             })
         });
-        test('GET /api/articles/:article_id/comments responds with \'Path Not Found\' message and 400 code, when requested with a malformed path', () => {
+        test('GET /api/articles/:article_id/comments returns 404, when article ID is not found', () => {
             return request(app)
-            .get("/api/articles/1/Komments")
-            .expect(400)
+            .get("/api/articles/8888/comments")
+            .expect(404)
             .then((response)=> {
-                expect(response.body.msg).toBe("Path Not Found")
+                expect(response.body.msg).toBe("No article found under ID: 8888")
+            })
+        });
+
+    describe('POST /api/articles/:article_id/comments', () => {
+        test('POST /api/articles/:article_id/comments, posts a new comment to our array of comments by article ID.', () => {
+            const body = "This is an added comment"
+            const userData = data.userData[0].username;
+
+            return request(app)
+            .post("/api/articles/6/comments")
+            .expect(200)
+            .send({ username: userData, body: body,})
+            .then((response)=>{
+                const postedComments = response.body.comment;
+                expect(postedComments).toHaveProperty("author");
+                expect(postedComments).toHaveProperty("body");
+            })
+        });
+        test('POST /api/articles/:article_id/comments, responds with a status code 400, when missing required fields in body', () => {
+    
+            return request(app)
+            .post("/api/articles/9/comments")
+            .expect(400)
+            .send({ })
+            .then((response)=>{
+                expect(response.body.msg).toBe("Bad Request");
+            })
+        });
+        test('POST /api/articles/:article_id/comments, responds with a status code of 404, when article ID is not found in db', () => {
+            const body = "This is an added comment"
+            const userData = data.userData[1].username;
+
+            return request(app)
+            .post("/api/articles/777/comments")
+            .expect(404)
+            .send({ username: userData, body: body })
+            .then((response)=>{
+                expect(response.body.msg).toBe("No article found under ID:777")
```

Comment:
Great tests here, though would be good to also add a test to check what happens when a username not in the database is used ✅